Ensuring users can truly pass options to transform.

antwan1986 opened the pull request:
Hello again,

I hope PR's are welcome; I was experimenting with JSPot a bit more after you very kindly updated the code base to handle ES2015 source files. During this experimenting I hit a problem where I needed to set more options for falafel / acorn that your current option (`ecmaVersion`), however, no other setting I applied seemed to be getting used.

I did some digging for an hour or so through your code base and realised that `transform` can sometimes be called without the options argument meaning at times it'll be called and ignore any settings you've supplied. `transform` seems to get called once without options right from launch and it seems to be `yoink_members` which does the calling.

You'll know the code base better than I do and the implications of my proposed changes, but in this PR I'm merely updating `yoink_members` method so it respects options that had been passed through to it. I'm also updating the `gather` method which was also passing options through to `transform` but wasn't extending any set by the user.

Have a look and see what you think!

Anthony

justinvdm commented:
Hi, thanks for the PR :) +1 on passing through parser options, this is probably something I should've added in #38.

I'd be a bit more comfortable with us scoping the options we pass to falafel to avoid name conflicts, maybe something like `{parser: {/* falafel options go here */}}`?

justinvdm commented:
Two other things I think we might need for this:
  - If you would like to use this feature via the cli, we'll need to add a way of passing through parser options via the cli. If you don't need the cli, then I think we can leave this for now.
  - It would be great if we could add a test for supporting passing through parser options. Maybe there is some falafel or acorn option we could pass through in a test that would make testing this easy enough?

antwan1986 commented:
Hey!

Personally I don't need it via the CLI, I'm doing everything via a little CommonJS / Gulp Task I've written. As for the isolation of the falafel options into an object I think that'd be a great idea. I can certainly update the test suite with something to ensure that falafel options are accepted and indeed extended.

I'll try to get something written over the next couple of days!

antwan1986 commented:
Have a look at what I've just committed. I've moved the Acorn / Falafel options (`ecmaVersion`, `locations`) into an object to encapsulate them, I've given the ability for the user to overwrite said object and I've also updated the test suite to ensure this.

Hopefully this all aligns with what you were imagining!

justinvdm commented:
Looks great :+1:, will merge and do a release in a bit.

justinvdm commented:
Released as v0.3.10

antwan1986 commented:
Excellent! Thanks so much for all the support.
